### DISCUSSION TITLE
About commit message customization

### DISCUSSION THREAD

**Discussion by ShingZhanho** [JamesIves/github-pages-deploy-action#612] [Ideas]
Hi there! I know that this action supports commit message customization, but how? Are there any placeholders that can let me customize the message? I've noticed the issue #133 :

> Right now `Deploying to {branchname} - {sha}` is fine for 90% of cases........

but I'm not sure if `{branchname}` and `{sha}` are actually usable placeholders.

**Comment by JamesIves** [comment]
You're able to customize the commit message using the 'commit-message' input. Currently you're not able to utilize any form of placeholder replacement though. What are you trying to achieve?

**Reply by ShingZhanho** [reply]
I want the commit message to be something like the following to meet my commit message convention:
```
gh-pages, docs: auto-deploy to github pages @ abcdef
gh-pages, docs: auto-deploy to github pages @ 1a2b3c
...
```
where `abcdef` and `1a2b3c` is the sha and is different each time.

**Reply by JamesIves** [reply]
Seems like a reasonable request to me. If you'd like to submit a [pull request](https://github.com/JamesIves/github-pages-deploy-action/blob/dev/CONTRIBUTING.md) for this I'd be happy to review it. I've done something similar to this using mustache.js [in another action I maintain if you'd like to see an example](https://github.com/JamesIves/fetch-api-data-action/blob/dev/src/fetch.ts#L25).

**Reply by ShingZhanho** [reply]
I am not familiar with javascript so I'm afarid I can't contribute to your project. 😬 Btw thank you for replying me!
